Drivers/NER/src/cascadiamc.c

```diff
@@ -0,0 +1,407 @@
+            mc->throttle_signal = message.data >> 56;
+            mc->brake_signal = ((message.data >> 48) & 0x0000000000FF);
+            mc->drive_enable = ((message.data >> 39) & 0x000000000001);
+    }
+}
+
+int32_t cascadiamc_get_rpm(cascadiamc_t* mc)
+{
+    return mc->rpm;
+}
+
+int16_t cascadiamc_get_duty_cycle(cascadiamc_t* mc)
+{
+    return mc->duty_cycle;
+}
+
+int16_t cascadiamc_get_input_voltage(cascadiamc_t* mc)
+{
+    return mc->input_voltage;
+}
+
+int16_t cascadiamc_get_ac_current(cascadiamc_t* mc)
+{
+    return mc->ac_current;
+}
+
+int16_t cascadiamc_get_dc_current(cascadiamc_t* mc)
+{
+    return mc->dc_current;
+}
+
+int16_t cascadiamc_get_controller_temp(cascadiamc_t* mc)
+{
+    return mc->cont_temp;
+}
+
+int16_t cascadiamc_get_motor_temp(cascadiamc_t* mc)
+{
+    return mc->motor_temp;
+}
+
+uint8_t cascadiamc_get_fault_code(cascadiamc_t* mc)
+{
+    return mc->fault_code;
+}
+
+int8_t cascadiamc_get_throttle_signal(cascadiamc_t* mc)
+{
+    return mc->throttle_signal;
+}
+
+int8_t cascadiamc_get_brake_signal(cascadiamc_t* mc)
+{
+    return mc->brake_signal;
+}
+
+int8_t cascadiamc_get_drive_enable(cascadiamc_t* mc)
+{
+    return mc->drive_enable;
+}
+
+void cascadiamc_set_current(int16_t current)                                /* SCALE: 10          UNITS: Amps       */
+{
+    union
+    {
+        uint8_t msg[8] = {0, 0, 0, 0, 0, 0, 0, 0};
+
+        struct
+        {
+            uint16_t ac_current;
+        } cfg;
+
+    } set_current_data;
+
+    uint32_t msg_id = 0x00000110;
+    uint8_t msg_len = 8;
+    uint8_t msg_line = 1;
+    set_current_data.cfg.ac_current = current;
+    can_msg_t current_msg;
+    current_msg.data = set_current_data.cfg;
+    current_msg.line = msg_line;
+    current_msg.len = msg_len;
+    current_msg.id = msg_id;
+    
+    CAN_StatusTypedef status = can_send_message(current_msg);
+}
+
+void cascadiamc_set_brake_current(int16_t brake_current)                    /* SCALE: 10          UNITS: Amps       */
+{
+    union
+    {
+        uint8_t msg[8] = {0, 0, 0, 0, 0, 0, 0, 0};
+
+        struct
+        {
+            uint16_t brake_current;
+        } cfg;
+
+    } set_brake_current_data;
+
+    uint32_t msg_id = 0x00000210;
+    uint8_t msg_len = 8;
+    uint8_t msg_line = 1;
+    set_brake_current_data.cfg.brake_current = brake_current;
+    can_msg_t brake_current_msg;
+    brake_current_msg.data = set_brake_current_data.cfg;
+    brake_current_msg.line = msg_line;
+    brake_current_msg.len = msg_len;
+    brake_current_msg.id = msg_id;
+    
+    CAN_StatusTypedef status = can_send_message(brake_current_msg);
+}                                                  
+
+void cascadiamc_set_speed(int32_t rpm)                                      /* SCALE: 1           UNITS: RPM        */
+{
+    union
+    {
+        uint8_t msg[8] = {0, 0, 0, 0, 0, 0, 0, 0};
+
+        struct
+        {
+            uint32_t erpm;
```

Comment:
just a q, what does the "e" in ``erpm`` stand for

Reply:
electrical